code/script.js

```diff
@@ -220,106 +225,160 @@ const generateBoard = () => {
       </div>
     `
   })
+
+  board.innerHTML = ''
+  charactersInPlay.forEach((person) => {
+    board.innerHTML += `
+      <div class="card">
+        <p>${person.name}</p>
+        <img src=${person.img} alt=${person.name}>
+        <div class="guess">
+          <span>Guess on ${person.name}?</span>
+          <button class="filled-button small" onclick="guess('${person.name}')">Guess</button>
+        </div>
+      </div>
+    `
+  })
 }
 
-// Randomly select a person from the characters array and set as the value of the variable called secret
+// A person is randomly selected a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
-}
+};
 
-// This function to start (and restart) the game
+// This function starts (and restarts) the game
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
+
+  // By invoking below functions, a new board of characters is generated and a secret person selected. 
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
-}
+  generateBoard();
+  setSecret() ;
+};
 
-// setting the currentQuestion object when you select something in the dropdown
+// Setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+const category = questions.options[questions.selectedIndex].parentNode.label
+const value = questions.value; // This variable stores what option group (category) the question belongs to.
```

Comment:
I'm not sure if it's necessary to use semicolons (for example see line above where you don't use one. But I think it's ok as long as there is consistency with it.